proxy/src/telemetry/metrics/mod.rs

```diff
@@ -65,6 +66,15 @@ use self::labels::{
 pub use self::labels::DstLabels;
 pub use self::record::Record;
 
+trait FmtMetric {
+    fn fmt_metric<N: Display>(&self, f: &mut fmt::Formatter, name: N) -> fmt::Result;
```

Comment:
Is `fmt_metric` ever actually used for anything besides the `process_start_time` metric? 

It seems like everything else only uses `fmt_metric_labeled`. Maybe we could remove this method and just give the `process_start_time` metric a special `fmt_metric_labeled` impl that just ignores any passed-in labels. Up to you whether that's better or not.

Reply:
more generally, i don't think a given metric type (i.e. Counter/Gauge/Histogram) makes sense in only labeled or non-labeled contexts. I think the alternative would be to add another trait

```
trait FmtLabels: Display {
    fn is_empty(&self) -> bool;
}
```

but i opted to not introduce yet another trait

Reply:
Fair enough; I thought it might be preferable to avoid merging in what's essentially dead code and add it when it becomes necessary, but I suppose special-casing the `process_start_time` metric for now is messier.